crates/sui-core/src/validator_client_monitor/stats.rs

```diff
@@ -127,149 +128,73 @@ impl ClientObservedStats {
-                };
-                if let Some(max_latency) = max_latencies.get(&op) {
-                    if latency > *max_latency {
-                        max_latencies.insert(op, latency);
-                    }
-                } else {
-                    max_latencies.insert(op, latency);
-                }
-            }
-        }
-
-        max_latencies
-    }
-
-    /// Calculate client-observed score for a single validator for the provided tx type.
+    /// Calculate latency-based latency for a single validator for the provided tx type.
     ///
-    /// The score combines reliability and latency metrics as observed by the client,
-    /// weighted according to configuration.
+    /// Returns the average latency for relevant operations (Consensus and FastPath only)
+    /// with reliability penalty applied. Lower values are better.
     ///
-    /// If a validator is missing local stats for an operation type, we use a
-    /// conservative default (assuming they are at the global maximum latency)
-    /// to ensure fairness while still allowing them to be scored.
+    /// Only considers:
+    /// - Consensus operations (for SharedObject transactions)
+    /// - FastPath operations (for SingleWriter transactions)
     ///
-    /// Score calculation:
-    /// 1. Latency scores are normalized against global maximums
-    /// 2. Each operation type has its own weight
-    /// 3. Final score = (weighted_latency_score * latency_weight) + (reliability * reliability_weight)
-    fn calculate_client_score(
-        &self,
-        stats: &ValidatorClientStats,
-        max_latencies: &HashMap<OperationType, f64>,
-        tx_type: TxType,
-    ) -> f64 {
-        let mut latency_score = 0.0;
-        let mut total_weight = 0.0;
+    /// Returns latency in seconds, with reliability penalty applied as a multiplier.
+    fn calculate_client_latency(&self, validator: &AuthorityName, tx_type: TxType) -> f64 {
+        let mut final_latency = MAX_LATENCY;
 
-        for op in OperationType::iter() {
-            let latency_weight = match op {
-                OperationType::Submit => self.config.score_weights.submit_latency_weight,
-                OperationType::Effects => self.config.score_weights.effects_latency_weight,
-                OperationType::HealthCheck => self.config.score_weights.health_check_latency_weight,
-                OperationType::FastPath => self.config.score_weights.fast_path_latency_weight,
-                OperationType::Consensus => self.config.score_weights.consensus_latency_weight,
-            };
+        let Some(stats) = self.validator_stats.get(validator) else {
+            return final_latency;
+        };
 
-            if tx_type == TxType::SingleWriter && op == OperationType::Consensus {
-                continue;
+        if let Some(exclusion_time) = stats.exclusion_time {
+            if exclusion_time.elapsed() < self.config.failure_cooldown {
+                return final_latency;
```

Comment:
nit: I don't think we should track what `final_latency` is.
```suggestion
                return MAX_LATENCY;
```